modules/spo-memory-profiling.adoc

```diff
@@ -0,0 +1,64 @@
+// Module included in the following assemblies:
+//
+// * security/security_profiles_operator/spo-troubleshooting.adoc
+
+:_content-type: PROCEDURE
+[id="spo-memory-profiling_{context}"]
+= Enable CPU and memory profiling
+
+You can enable the CPU and memory profiling endpoints for debugging purposes.
+
+.Procedure
+
+. To use the profiling support, patch the `spod` configuration and set the `enableProfiling` value by running the following command:
++
+[source,terminal]
+----
+$ oc -n openshift-security-profiles patch spod \
+    spod --type=merge -p '{"spec":{"enableProfiling":true}}'
+----
++
+.Example output
+[source,terminal]
+----
+securityprofilesoperatordaemon.security-profiles-operator.x-k8s.io/spod patched
+----
+
+. Verify the `openshift-security-profiles` container is serving the profile endpoint by running the following command:
++
+[source,terminal]
+----
+$ oc logs --selector name=spod -c openshift-security-profiles | grep "Starting profiling"
+----
++
+.Example output
+[source,terminal]
+----
+I1202 15:14:40.276363 2185724 main.go:226]  "msg"="Starting profiling server"  "endpoint"="localhost:6060"
+----
+
+. Verify the `log-enricher` container is serving the profile endpoint by running the following command:
++
+[source,terminal]
+----
+$ oc logs --selector name=spod -c log-enricher | grep "Starting profiling"
```

Comment:
The container in the command is wrong.
@jhrozek Does it work? I didn't see expected result with the right command.
$  oc logs --selector name=spod -c openshift-security-profiles | grep "Starting profiling"
error: container openshift-security-profiles is not valid for pod spod-8cbll

$  oc logs --selector name=spod -c security-profiles-operator | grep "Starting profiling"

Reply:
FWIW, @xiaojiey pinged me about this and I've been having issues getting my clusters to work properly today, so I haven't been able to reproduce or test this section.

In terms of prioritization, I'd say give me one more day to test things out -- if the profiling doesn't work properly, we'd just remove or comment out this section. Profiling is meant as a tool to help support investigate scaling or performance issues and wouldn't be a GA blocker (IMO) anyway.

Reply:
Hmm, it turns out that the upstream docs didn't reflect some changes in the code done upstream after the fact. The profiling works, but a bit differently than advertised in the upstream docs (and kindof by extension in the downstream docs as well).

@sheriff-rh @xiaojiey I suggest we disable out this module for the time being, open a bug against SPO to get the docs fixed upstream and then with the next update, fix the downstream docs and enable this module again. If a customer run into a performance issue that needs to be profiled, we can advise them on how to do that in the meantime.

Reply:
If anyone absolutely needs profiling, then the way to check it is actually:
```
oc describe ds spod | grep PROFILING
      SPO_PROFILING:       true
      SPO_PROFILING_PORT:  6060
```
This says the profiling endpoint listens on 6060, then we can curl the endpoint:
```
curl http://10.128.0.83:6060/debug/pprof/heap
Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.
```
and we'd save the result into a file and pass that file into `go tool pprof`. But I think we should fix the upstream docs first.

Reply:
Let's comment it out first and add it back as intended later.